File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2020/ID06TC1/BridgeTest/BridgeTest/FigureFabric.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgeTest
{
    delegate Figure FigureMaker(int[] Parameters);
    public class FigureFabric
    {
        private static readonly Dictionary<String, FigureMaker> map = new Dictionary<string, FigureMaker>()
        {
            {"Circle",(Parameters) => new FigureCircle(Parameters)},
            {"Triangle",(Parameters) => new FigureTriangle(Parameters)},
            {"Rectangle",(Parameters) => new FigureRectangle(Parameters)},
        };
        public static string[] Kinds
        {
            get { return map.Keys.OrderBy((s) => s).ToArray(); }
        }
        public static Figure make(string Kind, string Parameters)
        {
            string[] items = Parameters.Split(" \t,;".ToArray(), StringSplitOptions.RemoveEmptyEntries);
            int[] tb = new int[items.Length];
            for (int i = 0; i < tb.Length; ++i) tb[i] = int.Parse(items[i]);
            return map[Kind](tb);
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/Java/2020/BridgeTest/src/FigureFabric.java

import java.util.HashMap;
import java.util.List;
import java.util.stream.Collectors;

public class FigureFabric
{
   private static final HashMap<String,FigureMaker> map=new HashMap<String,FigureMaker>()
   {
      {
         put("Circle",(tb) -> new FigCircle(tb));
         put("Rectangle",(tb) -> new FigRectangle(tb));
         put("Triangle",(tb) -> new FigTriangle(tb));
      }
   };
   public static List<String> keys()
   {
      return map.keySet().stream().sorted().collect(Collectors.toList());
   }
   public static Figure make(String kind, String parameters)
   {
      String[] param=parameters.split("[\\s\t,;]+");
      int[] tb=new int[param.length];
      for(int i=0;i<tb.length;++i) tb[i]=Integer.parseInt(param[i]);
      return map.get(kind).make(tb);
   }
}
